### ISSUE TITLE
New Model: PialNN/pialnn/1.0.0

### ISSUE THREAD

**Issue by hvgazula** [gaiborjosue/trained-models-fork#64]
### Model's repository path

PialNN/pialnn/1.0.0

### Best model weights

https://drive.google.com/uc?id=18KdBZLwd-8ld3SZoQCXAXwvrCtdeABU-&confirm=t

### Docker information

https://github.com/gaiborjosue/DeepCSR-testing/tree/main/dockerP

### Model's python scripts

https://github.com/gaiborjosue/DeepCSR-testing/blob/main/pial/predict.py
https://github.com/gaiborjosue/DeepCSR-testing/blob/main/pial/utils.py
https://github.com/hvgazula/pialnn-forked/tree/harsha/data
https://github.com/hvgazula/pialnn-forked/tree/harsha/model
https://github.com/gaiborjosue/DeepCSR-testing/blob/main/pial/config.py

### Model's card and spec

https://github.com/gaiborjosue/trained-models-fork/blob/master/DeepCSR/deepcsr/1.0/spec.yaml
https://github.com/gaiborjosue/trained-models-fork/blob/master/DeepCSR/deepcsr/1.0/model_card.yaml

### Sample data

https://hvgazula.github.io/pialnn-forked/sample.zip

### Test Command

python ./PialNN/pialnn/1.0.0/predict.py --model ./PialNN/pialnn/1.0.0/weights/pialnn_model_lh_200epochs.pt --data_path=./PialNN/pialnn/1.0.0/data/test/ --hemisphere=lh --save_mesh_eval=False

### Read & Followed recommendations

- [X] I have read and followed the recommedations in this form and have assigned myself to the issue.

**Comment by hvgazula** [comment]
Create the add model issue but could not assign a label. 🤷‍♂️

**Comment by gaiborjosue** [comment]
@hvgazula It does not let you assign a label either?

**Comment by hvgazula** [comment]
No. Because I am not a collaborator on this repo?

**Comment by gaiborjosue** [comment]
Damn. Maybe. Let me research this.

**Comment by gaiborjosue** [comment]
Indeed. People have to be collaborators in the repo to be able to do that smh

**Comment by gaiborjosue** [comment]
A possible solution came to my mind @hvgazula. What if instead of assigning labels, users comment on the issue the label? The workflow will get triggered, and the bot will be the one changing the labels to whatever the user comments.

So first, the issue will automatically have "Ready-to-test" as a default label. If success, then nothing should be done because the action will just tag the issue with "success." However, if failed, the issue will be tagged as "failed" and the user must comment on the issue "Ready-to-test" for the bot to change the tag of the issue and make the workflow rerun again.

**Comment by hvgazula** [comment]
So, do I create a new issue again or just comment whenever your changes are ready?

**Comment by gaiborjosue** [comment]
For your testing with the issue of PialNN don't worry, I will assign a label. That will give me some time to do the updates 😅

**Comment by gaiborjosue** [comment]
Or as you said I will let you know when the changes are ready, so we can test them out with your issue. Now I think this is a better idea 😅

**Comment by hvgazula** [comment]
Please do. I will be keeping an eye on the GH workflow. I will work on cortexode in the meantime. Besides that anything else for me.

**Comment by gaiborjosue** [comment]
Thanks. No, just trying to think of a better approach to this issue 🤔

**Comment by gaiborjosue** [comment]
@hvgazula docker information folder does not exist

**Comment by gaiborjosue** [comment]
@hvgazula Sample zip data should be in google drive url

**Comment by hvgazula** [comment]
> @hvgazula docker information folder does not exist

Please change `eg_testing` to `harsha`.

**Comment by hvgazula** [comment]
> @hvgazula Sample zip data should be in google drive url

What do you mean 'should be'? The instruction only says 'direct download' and I gave you that.

**Comment by gaiborjosue** [comment]
In the instructions/docs it says "- Upload a Google Drive direct download URL pointing to the sample dataset used to test the model. Follow the URL structure: ```"https://drive.google.com/uc?id=<FILEID>&confirm=t"```"

**Comment by gaiborjosue** [comment]
Also, the test command is not using relative paths to the repo, it just has the files and that is it. It should follow the structure of DeepCSR's test command:

python ./Testing/test/1.0/predict.py --conf_path ./Testing/test/1.0/config/predict.yaml --model_checkpoint ./Testing/test/1.0/weights/best_model.pth --dataset ./Testing/test/1.0/example-data/sample.nii.gz --output_dir .

Instead of "Testing" and "test" and "1.0" you should enter your info.

Weights are under the folder "weights", datasets are under the folder "example-data"

And you should please add a "." before each path, since it is relative

**Comment by hvgazula** [comment]
Yes, please add it yourself for the time being as you have the zip file anyway. I am working on finding a different model for testing.

**Comment by gaiborjosue** [comment]
I am testing the zip right now with deepcsr. And I get this error:

https://github.com/gaiborjosue/trained-models-fork/actions/runs/6511766055/job/17688329367#step:11:25

**Comment by gaiborjosue** [comment]
Spec yaml info:

https://github.com/neuronets/trained-models/blob/d96e8da11e0c8c4f4c955931fa424ed6e433b730/docs/spec.yaml

https://github.com/gaiborjosue/trained-models-fork/blob/master/DDIG/SynthMorph/1.0.0/brains/spec.yaml

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by hvgazula** [comment]
Please paste the errors here. Easy for me to help. I am working on other models right now.

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Branch issue-64 created for issue: New Model: PialNN/pialnn/1.0.0

**Comment by gaiborjosue** [comment]
Could not create PR (Resource not accessible by personal access token)

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
[This is an automated message] Hello @hvgazula. The folders/scripts you provided did not pass our tests. Please check out the Action, and modify this issue accordingly with other/same urls. 

When ready, just change the label of this issue to "Ready for testing".

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by hvgazula** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Hello @hvgazula it wont work since the dockerfile is using "pialnn" as conda env we have to switch it to base, just as we did with DeepCSR

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by hvgazula** [comment]
> Hello @hvgazula it wont work since the dockerfile is using "pialnn" as conda env we have to switch it to base, just as we did with DeepCSR

If you changed it yourself somewhere, can you please point me to the updated docker file so I can go ahead and start testing the CortexODE model?

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by gaiborjosue** [comment]
Ready-to-test

**Comment by hvgazula** [comment]
Okay.. this is good news. Can you tell me what all things you had to change from the point I uploaded to the point it ran successfully?

Of course, I know my command is wrong but I think I gave the (nearly) correct one as well that excludes the paths. I'll be looking forward to the notes. Congrats!!! 👍  :)
